django/forms/models.py

```diff
@@ -1009,9 +1009,15 @@ def _get_foreign_key(parent_model, model, fk_name=None, can_fail=False):
         fks_to_parent = [f for f in opts.fields if f.name == fk_name]
         if len(fks_to_parent) == 1:
             fk = fks_to_parent[0]
-            if not isinstance(fk, ForeignKey) or \
-                    (fk.remote_field.model != parent_model and
-                     fk.remote_field.model not in parent_model._meta.get_parent_list()):
+            remote_field_is_proxy = fk.remote_field and fk.remote_field.model._meta.proxy
+            if not isinstance(fk, ForeignKey) or (
+                not remote_field_is_proxy and
+                fk.remote_field.model != parent_model and
+                fk.remote_field.model not in parent_model._meta.get_parent_list()
+            ) or (
+                remote_field_is_proxy and
+                fk.remote_field.model._meta.proxy_for_model not in parent_model._meta.get_parent_list()
+            ):
```

Comment:
In both cases we're checking `fk.remote_field.model._meta.proxy` when `fk` is a `ForeignKey` so there is no need to check `fk.remote_field`. I would also add `parent_list` variable and small comments:
```suggestion
            parent_list = parent_model._meta.get_parent_list()
            if not isinstance(fk, ForeignKey) or (
                # ForeignKey to proxy models.
                fk.remote_field.model._meta.proxy and 
                fk.remote_field.model._meta.proxy_for_model not in parent_list
            ) or (
                # ForeignKey to concrete models. 
                not fk.remote_field.model._meta.proxy and
                fk.remote_field.model != parent_model and
                fk.remote_field.model not in parent_list
            ):
```